lib/install/management/store_files.go

```diff
@@ -53,15 +53,37 @@ func (d *Dispatcher) DeleteStores(vchVM *vm.VirtualMachine, conf *metadata.Virtu
 	var emptyImages bool
 	var emptyVolumes bool
 	log.Infof("Removing images")
-	if emptyImages, err = d.deleteImages(ds, p); err != nil {
-		errs = append(errs, err.Error())
+	for _, imageDir := range conf.ImageStores {
+		imageDSes, err := d.session.Finder.DatastoreList(d.ctx, imageDir.Host)
```

Comment:
@hickeng at this point in the code we already have the URL extracted and stored. It's extracted using a helper method, but not one of Faiyaz's (which actually I wrote the majority of) because Faiyaz's code goes from URL format to vSphere datastore formate( `ds://host/path`  to `[host] path` ) as strings on either end or "[host] path" string to `url.URL` -- the helper that goes from `ds://host/path` (string) to url.URL is a vic-machine specific one -- the one you helped me edit (`dsHelper` I believe is the name) so I'm not entirely sure to what you're referring in this case.